### DISCUSSION TITLE
get market or identifier for candle response event

### DISCUSSION THREAD

**Discussion by csidi90** [altangent/ccxws#266] [Q&A]
Hi, so I have a problem which I can not solve and hope some of you can answer.

I have a ccxws socket that subscribed to multiple markets on binance. All is working fine but I have big troubles identifying the corresponding market from the candle response as it just contains timestamp and ohlcv data, but no data for which market / symbol it actually is. I need to build caching for all markets and without knowing to which market the response event belonngs, I can not cache the data.

Regards

**Comment by csidi90** [comment]
Nevermind I was to stupid too read that the.on accepts a market field as a second param. My bad

**Reply by bmancini55** [reply]
No worries! Thanks for asking the question.  I added an example below.

**Comment by bmancini55** [comment]
All data events (`ticker`, `candle`, `trade`, `l2update`, `l2snapshot`', etc) emit the market as the second parameter in the handler function.  For example:

```javascript
client.subscribeCandles(market);
client.on('candle', (candle, market) => {
  // do something with candle and market
});
```
